src/sonic-eventd/src/eventd.cpp

```diff
@@ -409,8 +409,30 @@ capture_service::do_capture()
          *
          * This behavior will only happen once when XSUB connects to XPUB not everytime cache is started.
          *
+         * There are chances that there are events already published to XSUB endpoint before XSUB is able to connect to XPUB, so we can receive events
+           before the subscription message
          */
-         init_done = true;
+
+
+        if(rc == 1) { // Expected case to receive subscription message as very first message
+            SWSS_LOG_INFO("Received subscription message when XSUB connects to XPUB");
```

Comment:
If there are too much "received subscription message" log, suggest change to SWSS_LOG_DEBUG.

Reply:
This will happen only once during process lifetime.

Reply:
#Closed